File: aoc2020/day2/part2.py
def main():
    with open("./input.txt", "r") as f:
        p = f.read().split("\n")

    countValidPass = 0

    for item in p:
        policy, passw = item.split(":")
        count, char = policy.split(" ")
        lowC, upperC = count.split("-")

        if passw[int(lowC) - 1] == char and passw[int(upperC) - 1] != char:
            countValidPass += 1
        elif passw[int(lowC) - 1] != char and passw[int(upperC) - 1] == char:
            countValidPass += 1
    print(countValidPass)


if __name__ == "__main__":
    main()

File: aoc2020/day2/part2.ts
import * as fs from "node:fs/promises";

async function main() {
  const data = (await fs.readFile("./input.txt", "utf-8")).trim().split("\n");
  let countValidPass = 0;
  for (const line of data) {
    let [policy, passw] = line.split(":");
    let [count, char] = policy.split(" ");
    let [lowC, upperC] = count.split("-");

    let charCount = 0;
    for (const c of passw) {
      if (c == char) {
        charCount += 1;
      }
    }
    if (charCount >= Number(lowC) && charCount <= Number(upperC)) {
      countValidPass += 1;
    }
  }
  console.log(countValidPass);
}

await main();
